src/bin/dsd-analyze.rs

```diff
@@ -21,20 +21,14 @@ fn main() -> io::Result<()> {
     init_logging();
     let args = Args::parse();
 
-    let bytes: Bytes = if let Some(input_file) = args.input {
+    let mut reader = if let Some(input_file) = args.input {
         let file_path = Path::new(&input_file);
 
-        Bytes::from(fs::read(file_path).unwrap())
+        DogStatsDReader::new(fs::File::open(file_path).unwrap()).unwrap()
```

Comment:
Mild suggestion to get `thiserror` involved and have the path in the error, or print it in an `except`.